### DISCUSSION TITLE
beego를 이용한 Validation 관련

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#752] [General]
현재 TB에는 beego를 이용하여 Validation 을 수행하는 코드가 있습니다.

`cb-tumblebug/src/api/rest/server/common/utility.go` 에 다음과 같은 함수가 정의되어 있습니다.

```Go
import (
	"github.com/beego/beego/v2/core/validation"
)

func Validate(c echo.Context, params []string) error {
	valid := validation.Validation{}

	for _, name := range params {
		valid.Required(c.Param(name), name)
	}

	if valid.HasErrors() {
		for _, err := range valid.Errors {
			return errors.New(fmt.Sprintf("[%s]%s", err.Key, err.Error()))
		}
	}
	return nil
}
```

그리고, 다른 함수에서 다음과 같이 활용합니다.

`cb-tumblebug/src/api/rest/server/common/namespace.go`

```Go
func RestCheckNs(c echo.Context) error {

	if err := Validate(c, []string{"nsId"}); err != nil {
		common.CBLog.Error(err)
		return SendMessage(c, http.StatusNotFound, err.Error())
	}
...
```

=> `"nsId"` 필드에 값이 채워져 있는지 검사

---

한편, 최근 이슈 #706 과 PR #717, #718 을 통해
`github.com/go-playground/validator` 를 이용하는 struct validation을 도입하였습니다.

---

사실 beego는 echo와 비슷한 Golang web framework로,
TB에서는 beego 중 validation package만 import하여 사용하고 있었습니다.

그런데, 현재 beego 패키지를 이용하여 수행 중인 "`"nsId"` 필드에 값이 채워져 있는지 검사" 는
최근 도입된 `github.com/go-playground/validator` 를 이용하면 더욱 간결하게 수행할 수 있을 것으로 보입니다.

beego 대신 `github.com/go-playground/validator` 를 사용하면
TB에서 사용 중인 validation framework를 하나로 통일하고 (꼭 통일해야 하는 것은 아니지만.. 😊)
dependency 목록에서 beego 를 삭제하여

- dependency 한 개를 줄이고
- 그럼으로써 security/vulnerability issue 발생 가능성도 줄이고
- 바이너리 사이즈를 약간 더 줄이는

등의 이점을 취할 수 있지 않을까 합니다.

@seokho-son 어떻게 생각하시나요? 😊

**Comment by seokho-son** [comment]
@jihoon-seo 매매매우 좋습니다.. ㅎㅎ

함수를 정리하는 내용을 이슈로 오픈해도 좋을 것 같습니다.. !!
